### ISSUE TITLE
一张H20显卡，预训练qwen3-8b模型显存溢出

### ISSUE THREAD

**Issue by liuzhipengchd** [modelscope/ms-swift#6329]
我的执行命令，max_length从8000 减少到了3000。
`
swift pt \
    --model /root/development/model/Qwen3-8B \
    --train_type full \
    --dataset '/root/development/data/chinese-c4#420000' \
              '/root/development/data/tfs-c4' \
    --torch_dtype bfloat16 \
    --streaming true \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 16 \
    --packing true \
    --eval_steps 500 \
    --save_steps 500 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --deepspeed zero3 \
    --max_length 3000 \
    --max_steps 10000 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --save_only_model true \
    --output_dir /root/development/output/train_1028_pt_8b \
    --attn_impl flash_attn
`

<img width="899" height="508" alt="Image" src="https://github.com/user-attachments/assets/8659522b-e660-400f-8227-e2753bff2fca" />

使用的是官方提供的3.8.5镜像

root@547e42b978a9:~/development# pip list | grep ms
aliyun-python-sdk-kms             2.16.5
ms-agent                          1.3.0
ms-opencompass                    0.1.6
ms-swift                          3.8.3
ms-vlmeval                        0.0.18
msgpack                           1.1.1
msgspec                           0.19.0

请问下，使用sft训练的时候，最大长度都可以，但是pt 就不行

**Comment by liuzhipengchd** [comment]
目前2个问题，需要大佬解答一下
1、目前去掉了deepseed 参数 可以正常训练
2、但是耗时有点长，有没有一个标准可以参考。
下面是 qwen3-8b模型 训练10000条数据，max_len 是3000的，batch_size 是1 （过大的话显存不够）

{'loss': 3.31534271, 'grad_norm': 11.5, 'learning_rate': 1.6e-06, 'token_acc': 0.41374097, 'epoch': 0.01, 'global_step/max_steps': '80/10000', 'percentage': '0.80%', 'elapsed_time': '8m 39s', 'remaining_time': '17h 53m 28s', 'memory(GiB)': 94.1, 'train_speed(iter/s)': 0.154017} 这个耗时正常吗

**Comment by slin000111** [comment]
> 目前2个问题，需要大佬解答一下 1、目前去掉了deepseed 参数 可以正常训练 2、但是耗时有点长，有没有一个标准可以参考。 下面是 qwen3-8b模型 训练10000条数据，max_len 是3000的，batch_size 是1 （过大的话显存不够）
> 
> {'loss': 3.31534271, 'grad_norm': 11.5, 'learning_rate': 1.6e-06, 'token_acc': 0.41374097, 'epoch': 0.01, 'global_step/max_steps': '80/10000', 'percentage': '0.80%', 'elapsed_time': '8m 39s', 'remaining_time': '17h 53m 28s', 'memory(GiB)': 94.1, 'train_speed(iter/s)': 0.154017} 这个耗时正常吗

`use_liger_kernel`试一下。
